src/filesplugin.js

```diff
@@ -1,388 +1,372 @@
-
-			fileList.fileActions.registerAction({
-				name: 'approval-approve',
-				displayName: (context) => {
-					if (context && context.$file) {
-						const state = context.$file.data('approvalState')
-						if (state === states.APPROVABLE) {
-							return t('approval', 'Approve')
-						}
-					}
-					return ''
-				},
-				mime: 'all',
-				order: -139,
-				iconClass: (fileName, context) => {
-					if (context && context.$file) {
-						const state = context.$file.data('approvalState')
-						if (state === states.APPROVABLE) {
-							return 'icon-checkmark'
-						}
-					}
-				},
-				permissions: OC.PERMISSION_READ,
-				actionHandler: (fileName, context) => { this.approveSingle(fileName, context, this) },
-			})
-
-			fileList.fileActions.registerAction({
-				name: 'approval-reject',
-				displayName: (context) => {
-					if (context && context.$file) {
-						const state = context.$file.data('approvalState')
-						if (state === states.APPROVABLE) {
-							return t('approval', 'Reject')
-						}
-					}
-					return ''
-				},
-				mime: 'all',
-				order: -139,
-				iconClass: (fileName, context) => {
-					if (context && context.$file) {
-						const state = context.$file.data('approvalState')
-						if (state === states.APPROVABLE) {
-							return 'icon-close'
-						}
-					}
-				},
-				permissions: OC.PERMISSION_READ,
-				actionHandler: (fileName, context) => { this.rejectSingle(fileName, context, this) },
-			})
-
-			fileList.fileActions.registerAction({
-				name: 'approval-request',
-				displayName: (context) => {
-					if (context && context.$file) {
-						const state = context.$file.data('approvalState')
-						if (state === states.NOTHING && OCA.Approval.userRules && OCA.Approval.userRules.length > 0) {
-							return t('approval', 'Request approval')
-						}
-					}
-					return ''
-				},
-				mime: 'all',
-				order: -139,
-				iconClass: (fileName, context) => {
-					if (context && context.$file) {
-						const state = context.$file.data('approvalState')
-						if (state === states.NOTHING && OCA.Approval.userRules && OCA.Approval.userRules.length > 0) {
-							return 'icon-approval'
-						}
-					}
-				},
-				permissions: OC.PERMISSION_READ,
-				actionHandler: this.request,
-			})
-
-			fileList.fileActions.registerAction({
-				name: 'approval-pending-info',
-				displayName: (context) => {
-					if (context && context.$file) {
-						const state = context.$file.data('approvalState')
-						if (state === states.PENDING) {
-							return t('approval', 'Show approval details')
-						}
-					}
-					return ''
-				},
-				mime: 'all',
-				order: -139,
-				iconClass: (fileName, context) => {
-					if (context && context.$file) {
-						const state = context.$file.data('approvalState')
-						if (state === states.PENDING) {
-							return 'icon-approval'
-						}
-					}
-				},
-				permissions: OC.PERMISSION_READ,
-				actionHandler: this.showPendingInfo,
-			})
+const inlineAction = new FileAction({
```

Comment:
Files actions should also theoretically be registered on init :)

Reply:
Also, for cleaner code, I suggest you create one js file per action, and import/register them all in the init file.
Will be much easier to navigate on which action is where and does what ;)